bundles/ui/org.eclipse.smarthome.ui.icon/src/main/java/org/eclipse/smarthome/ui/icon/internal/IconServlet.java

```diff
@@ -97,21 +102,62 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
             return;
         }
 
-        String filename = StringUtils.substringAfterLast(req.getRequestURI(), "/");
-
-        String iconName = StringUtils.substringBeforeLast(filename, ".");
+        String category = getCategory(req);
+        Format format = getFormat(req);
```

Comment:
if the UI sends the format, how does it now if there are icons with the according format?